### ISSUE TITLE
APQ: Initial request should have a status 404 (Not Found), otherwise proxies could cache it

### ISSUE THREAD

**Issue by jorisroling** [apollographql/apollo-server#7374]
I see the APQ initial request, returning:

```
{"errors":[{"message":"PersistedQueryNotFound","extensions":{"code":"PERSISTED_QUERY_NOT_FOUND"}}]}
```
with status code 200 (OK)

Should it not actually have a status code of 404? So that a proxy typicly will NOT cache it.

Currently (Apollo v4.3.3) it has a status code of 200 (as usual).
This results in caching Proxies caching it, thus always having APQ clients fall back to a second call with the actual query, and thus breaking the beauty of APQ with caching Proxies.

**Comment by glasser** [comment]
Do you find that this happens in practice? These errors should come with a `cache-control: private, no-cache, must-revalidate` header. Are those headers not showing up for you, or are your caches ignoring it?

**Comment by jorisroling** [comment]
Very true, I see that cache-control header indeed, but my caches seem to ignore this. 

I assumed wrongfully that my proxy was honouring the cache-control headers.

Thanks for getting back at me.

**Comment by glasser** [comment]
If there's another tag we can add to that field to make more caches respect it, that's definitely worth considering!

**Comment by jorisroling** [comment]
I solved my issue (for now) by intercepting expressjs res.sent, checking the cache-control headers already set, and if they are containg 'no-cache', respond with non-200 status.

I have also checked with the devs of the particular proxy (nuster) to understand why it ignores the cache-control header. It seems to be totally ignored, so an extra tag would have no impact here. But thanks for the suggestion!
